prov/sm2/src/sm2_ep.c

```diff
@@ -337,6 +337,68 @@ static ssize_t sm2_do_cma(struct sm2_ep *ep, struct sm2_region *peer_smr,
 	return FI_SUCCESS;
 }
 
+static int sm2_format_ipc(struct sm2_xfer_entry *xfer_entry, void *ptr,
+			  size_t len, enum fi_hmem_iface iface, uint64_t device)
+{
+	void *base;
+	int ret;
+	struct ipc_info *ipc_info = (struct ipc_info *) xfer_entry->user_data;
+
+	xfer_entry->hdr.proto = sm2_proto_ipc;
+	xfer_entry->hdr.size = len;
+	ipc_info->iface = iface;
+	ipc_info->device = device;
```

Comment:
I think you need to add delivery complete here.

Reply:
fixed